Client/Frontend/Library/HistoryPanel/HistoryPanel.swift

```diff
@@ -0,0 +1,651 @@
+    static let IconBorderWidth: CGFloat = 0.5
+    static let actionIconColor = UIColor.Photon.Grey40 // Works for light and dark theme.
+}
+
+private class FetchInProgressError: MaybeErrorType {
+    internal var description: String {
+        return "Fetch is already in-progress"
+    }
+}
+
+@objcMembers
+class HistoryPanel: UIViewController, LibraryPanel, Loggable, NotificationThemeable {
+    
+    // MARK: - Properties
+    
+    typealias HistoryPanelSections = HistoryPanelViewModel.Sections
+    typealias a11yIds = AccessibilityIdentifiers.LibraryPanels.HistoryPanel
+    
+    var libraryPanelDelegate: LibraryPanelDelegate?
+    var recentlyClosedTabsDelegate: RecentlyClosedPanelDelegate?
+    
+    let profile: Profile
+    let viewModel: HistoryPanelViewModel
+    private let clearHistoryHelper: ClearHistoryHelper
+
+    private var diffableDatasource: UITableViewDiffableDataSource<HistoryPanelSections, AnyHashable>?
+    private var hasRecentlyClosed: Bool { profile.recentlyClosedTabs.tabs.count > 0 }
+    
+    // UI
+    lazy private var tableView: UITableView = .build { [weak self] tableView in
+        guard let self = self else { return }
+        tableView.dataSource = self.diffableDatasource
+        tableView.addGestureRecognizer(self.longPressRecognizer)
+        tableView.accessibilityIdentifier = a11yIds.tableView
+        tableView.prefetchDataSource = self
+        tableView.delegate = self
+        tableView.register(TwoLineImageOverlayCell.self, forCellReuseIdentifier: TwoLineImageOverlayCell.reuseIdentifier)
+        tableView.register(TwoLineImageOverlayCell.self, forCellReuseIdentifier: TwoLineImageOverlayCell.accessoryUsageReuseIdentifier)
+        tableView.register(OneLineTableViewCell.self, forCellReuseIdentifier: OneLineTableViewCell.reuseIdentifier)
+        tableView.register(SiteTableViewHeader.self, forHeaderFooterViewReuseIdentifier: SiteTableViewHeader.reuseIdentifier)
+        
+        if #available(iOS 15.0, *) {
+            tableView.sectionHeaderTopPadding = 0
+        }
+    }
+    
+    lazy var longPressRecognizer: UILongPressGestureRecognizer = {
+        UILongPressGestureRecognizer(target: self, action: #selector(onLongPressGestureRecognized))
+    }()
+    
+    lazy var emptyStateOverlayView: UIView = createEmptyStateOverlayView()
+    var refreshControl: UIRefreshControl?
+    var clearHistoryCell: OneLineTableViewCell?
+    
+    // MARK: - Inits
+    
+    init(profile: Profile, tabManager: TabManager) {
+        self.clearHistoryHelper = ClearHistoryHelper(profile: profile, tabManager: tabManager)
+        self.viewModel = HistoryPanelViewModel(profile: profile)
+        self.profile = profile
+        
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    deinit {
+        browserLog.debug("HistoryPanel DEinited.")
```

Comment:
```suggestion
        browserLog.debug("HistoryPanel Deinitialized.")
```

I keep changing these but am seeing a pattern. Is there a convention I don't know about with "DEinited"?

Reply:
Nope it's just a preference. I felt like the caps on `DE` is eye catching in a log.